### PR TITLE
 Add TOC scroll position, toggle icon and style improvements

### PR DESCRIPTION
**Pull request by erangatl**
## Purpose
Add TOC scroll position, toggle icon and style improvements

### CODE REVIEW

File: en/theme/material/partials/nav.html [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+      </div>
+    {% endif %}
+    <ul class="md-nav__list" data-md-scrollfix>
+      {% for nav_item in nav %}
+        {% set path = "nav-" + loop.index | string %}
+        {% set level = 1 %}
+        {% include "partials/nav-item.html" %}
+      {% endfor %}
+    </ul>
+  </nav>
```
Comment (jeradrutnam) [inline en/theme/material/partials/nav.html]: Missing new line on file end

File: en/docs/assets/lib/backtotop/js/main.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+		offsetOpacity = 1200, //browser window scroll (in pixels) after which the "back to top" link opacity is reduced
+		scrollDuration = 700,
+		scrolling = false;
+
+	if( backTop ) {
+		//update back to top visibility on scrolling
+		window.addEventListener("scroll", function(event) {
+			if( !scrolling ) {
+				scrolling = true;
+				(!window.requestAnimationFrame) ? setTimeout(checkBackToTop, 250) : window.requestAnimationFrame(checkBackToTop);
```
Comment (jeradrutnam) [inline en/docs/assets/lib/backtotop/js/main.js]: Fix character length

File: en/docs/assets/lib/backtotop/js/main.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+		//smooth scroll to top
+		backTop.addEventListener('click', function(event) {
+			event.preventDefault();
+			(!window.requestAnimationFrame) ? window.scrollTo(0, 0) : Util.scrollTo(0, scrollDuration);
+		});
+	}
+
+	function checkBackToTop() {
+		var windowTop = window.scrollY || document.documentElement.scrollTop;
+		( windowTop > offset ) ? Util.addClass(backTop, 'cd-top--is-visible') : Util.removeClass(backTop, 'cd-top--is-visible cd-top--fade-out');
```
Comment (jeradrutnam) [inline en/docs/assets/lib/backtotop/js/main.js]: Fix character length

File: en/docs/assets/lib/backtotop/js/main.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+		( windowTop > offsetOpacity ) && Util.addClass(backTop, 'cd-top--fade-out');
+		scrolling = false;
+	}
+})();
```
Comment (jeradrutnam) [inline en/docs/assets/lib/backtotop/js/main.js]: Missing new line on file end

File: .gitignore [hunk 5]
```diff
@@ -21,3 +21,7 @@
 
 # virtual machine crash logs, see http://www.java.com/en/download/help/error_hotspot.xml
 hs_err_pid*
+
+.idea/
+*/.DS_Store
+en/site
```
Comment (jeradrutnam) [inline .gitignore]: File end should have a new line

File: en/docs/assets/lib/backtotop/js/main.js (hunk 3 above)
Comment (erangatl) [reply en/docs/assets/lib/backtotop/js/main.js]: Fixed

File: en/docs/assets/lib/backtotop/js/main.js (hunk 2 above)
Comment (erangatl) [reply en/docs/assets/lib/backtotop/js/main.js]: Fixed

File: en/theme/material/partials/nav.html (hunk 1 above)
Comment (erangatl) [reply en/theme/material/partials/nav.html]: Fixed

File: en/docs/assets/lib/backtotop/js/main.js (hunk 4 above)
Comment (erangatl) [reply en/docs/assets/lib/backtotop/js/main.js]: Fixed

File: .gitignore (hunk 5 above)
Comment (erangatl) [reply .gitignore]: Fixed

**Pull request merged**